Notes/Algorithms/Data Structures/Hash-Tables.c

```diff
@@ -1,230 +1,244 @@
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 #include <stdbool.h>
 
 #define SUCCESS 0
 #define ERROR -1
-#define OVERFLOW -2
-#define EMPTY -3
-#define TEST 9
-#define EXPECTED 10
-#define SIZE 10
 
-// Hash struct
+// Hash Item struct
 typedef struct Item {
-    int key;
+    char* key;
     int value;
 } Item;
 
-//create global testHash and expectedHash for testFunctions to use
-Item* testHash[SIZE];
-Item* expectedHash[SIZE];
+// Hash Table struct
+typedef struct HashTable {
+    int size;
+    Item** items;
+} HashTable;
 
-void traverse() {
-    for (int i = 0; i < SIZE; i++) {
-        int value = testHash[i]->value;
-        int key = testHash[i]->key;        
-        printf("Test Key: %d Value: %d \n", key, value);
-    }
-    for (int i = 0; i < SIZE; i++) {
-        int value = expectedHash[i]->value;
-        int key = expectedHash[i]->key;
-        printf("Expected Key: %d Value: %d \n", key, value);
+// returns true if a collision would happen for the initially hashed index for a key/value pair and false if no collision would occurred 
+bool isCollision(HashTable * table, int index) {
+    if (table->items[index] != NULL) {
+        return true;
     }
+    return false;
 }
 
-//compare checks the Items
-bool compare() {
-    for (int i = 0; i < SIZE; i++) {
-        int testKey = testHash[i]->key;
-        int testValue = testHash[i]->value;
-        int expectedKey = expectedHash[i]->key;
-        int expectedValue = expectedHash[i]->value;
+// compare checks the testHash and expectedHash for parity
+// if the two HashTables match returns true
+// if a mismatch occurs returns false
+bool compare(HashTable * testHash, HashTable * expectedHash) {
+    for (int i = 0; i < 10; i++) {
+        int testKey = testHash->items[i]->key;
```

Comment:
You are 100% correct. This didn't flag for me because it wasn't actually being used for comparing updated hashTables vs expected hashTables for the delete function. However instead of deleting I am keeping for use of implementing the test cases you suggested for the insert function. 
[commit 69e5bb796190a5641dca783ec30b9c96513030bb](https://github.com/kcarrel/Training-Notes/pull/12/commits/69e5bb796190a5641dca783ec30b9c96513030bb).